### ISSUE TITLE
create [leveldb] resource

### ISSUE THREAD

**Issue by ahdinosaur** [bigcompany/resources#66]
i think we should eventually move our persistence resource away from using JugglingDB and instead make resources for each individual database we wish to use, so to start i think LevelDB is a great choice for a key/value database as we can use it on both the server and the client.

https://github.com/rvagg/node-levelup

in order to be a proper persist resource, it needs to have a function that takes a resource and adds additional methods on it. so for LevelDB, we would add .create, .get, and .destroy. maybe we can implement .find and .all later using an additional library.

**Comment by jfhbrook** [comment]
Alternately, make a leveldb engine for jugglingdb.

**Comment by ahdinosaur** [comment]
yeah, that works too. i'm mostly concerned about trying to fit every database into the mold of JugglingDB, especially graph databases, but it does seem like people are working on it (https://github.com/1602/jugglingdb/issues/279) so we'll see what comes about.

i see no reason to break off support for JugglingDB for the existing forms of persist, but it would be nice if we have a standardized way to make new persist resources that for some reason or another didn't want to use JugglingDB. one possible way to do this is to move all the JugglingDB code into a jugglingdb resource and then use it as a fallback.

**Comment by ahdinosaur** [comment]
first pass at a possible solution to allow both new persistence resources and existing jugglingdb persistence: https://github.com/bigcompany/resources/commit/af92fd2859e5.

**Comment by ghost** [comment]
Really keen on this - i'm hoping to use OrientDB for a new project using big, and as it's a graph / doc / key-value store jugglingdb doesn't seem like a good fit.

**Comment by ghost** [comment]
Thanks for refactoring this, i'll try to create an orientdb resource using a similar structure to that of the JDB one.
